cmd/alertmanager/main.go

```diff
@@ -132,22 +138,29 @@ func main() {
 		nflog.WithMaintenance(15*time.Minute, stopc, wg.Done),
 		nflog.WithMetrics(prometheus.DefaultRegisterer),
 		nflog.WithLogger(logger.With("component", "nflog")),
-	)
+	}
+	if *meshListen == "" {
```

Comment:
It'd be clearer to conditionally append, rather than to remove the entry afterwards